src/encoded/types/file.py

```diff
@@ -332,6 +341,56 @@ def file_type(self, file_format, file_format_type=None):
     def superseded_by(self, request, superseded_by):
         return paths_filtered_by_status(request, superseded_by)
 
+    @calculated_property(
+        condition=show_google_transfer,
+        schema={
+            "title": "Google transfer",
+            "description": "Metadata required by Google Cloud to create transfer TSV.",
+            "comment": "Do not submit. Values are calculated from file metadata.",
+            "type": "object",
+            "notSubmittable": True,
+        }
+    )
+    def google_transfer(self, md5sum, file_size):
+        try:
+            external = self._get_external_sheet()
+        except HTTPNotFound:
+            return {}
+        conn = boto3.client('s3', config=Config(
+            signature_version=botocore.UNSIGNED,
+        ))
+        location = conn.generate_presigned_url(
+            ClientMethod='get_object',
+            Params={
+                'Bucket': external['bucket'],
+                'Key': external['key']
+            },
+            ExpiresIn=0
+        )
+        return {
+            'url': location,
+            'md5sum_base64': base64.b64encode(bytes.fromhex(md5sum)).decode("utf-8"),
+            'file_size': file_size
+        }
+
+    @calculated_property(
+        condition=lambda status=None: status in File.public_s3_statuses,
+        schema={
+            "title": "S3 URI",
+            "description": "The S3 URI of public file object.",
+            "comment": "Do not submit. Value is calculated from file metadata.",
+            "type": "string",
+            "notSubmittable": True, 
+        }
+    )
+    def s3_uri(self):
+        try:
+            external = self._get_external_sheet()
+        except HTTPNotFound:
+            return ''
```

Comment:
I do wish there was a better option here than just returning an empty string if the file doesn't have an s3 external sheet (there are a few, for whatever reason). Returning None doesn't seem right because I think that would show up in the JSON.

Reply:
`if you don't return at all - would not that mean that the property will not be there?`

All Python functions return None if there isn't an explicit return value. Not sure if the schema is smart enough to remove this field from JSON though, or use None as the value.

`would not that be the desired result for files without s3 external sheet?`
My thinking was that it would be more important to be consistent in having that field in all released/archived/revoked files for people who are expecting that field to be there, even if the value is blank.

`how are restricted files handled?`

This is a good point. Restricted files probably shouldn't calculated these properties, and actually the set_status endpoint for files should probably not try to set the ACL to public when these are released (separate ticket).

Reply:
The schema is "smart " enough to not put None in calculated values